sdk-lib/src/main/java/cash/z/ecc/android/sdk/Synchronizer.kt

```diff
@@ -333,6 +333,17 @@ interface Synchronizer {
         proposal: Proposal
     ): Pczt
 
+    /**
+     * Redacts information from the given PCZT that is unnecessary for the Signer role.
+     *
+     * @param pczt The partially created transaction in its serialized format.
+     *
+     * @return the updated PCZT in its serialized format.
+     *
+     * @throws PcztException.RedactPcztForSignerException as a common indicator of the operation failure
+     */
+    suspend fun redactPcztForSigner(pczt: Pczt): Pczt
+
     /**
```

Comment:
Does the ability to check whether the PCZT requires Sapling proofs also need to be exposed here, so that the wallet can ensure that it has downloaded the Sapling parameters ahead of time instead of failing when it tries to make the signatures?

Reply:
Yes; I ended up adding this to the Swift SDK today, and I'll go back and add it here.